hcl2template/types.packer_config.go

```diff
@@ -225,6 +225,21 @@ func (c *PackerConfig) parseLocalVariables(f *hcl.File) ([]*LocalBlock, hcl.Diag
 	return locals, diags
 }
 
+func (c *PackerConfig) evaluateAllLocalVariables(locals []*LocalBlock) hcl.Diagnostics {
+	var local_diags hcl.Diagnostics
+
+	// divide by 2 so that you don't get duplicate locals
+	// appear to have double locals in LocalBlock, not sure if intentional
+	for i := 0; i < len(locals)/2; i++ {
```

Comment:
eh nice fix and nice finding, these duplicate values should not be duplicate, I'll try to find the root cause.

Reply:
Ill keep taking a look too, was surprising when debugging. Even without the duplicates though, I think something like this is required to print all local errors if there is a circular error.

Reply:
Found the root cause, it's hapenning here: 

We first copy this value: 
https://github.com/hashicorp/packer/blob/91a0f97b32cce0e85b4da8e8cc5686ba0ef57f17/hcl2template/types.packer_config.go#L189

Then return it, but the returned value is also added to that c.LocalBlocks slice, it was introduced here: https://github.com/hashicorp/packer/pull/10509

Okay, but I think yours is good to go, and I'll fix the duplicate block issue right up next.